linkerd/io/src/sensor.rs

```diff
@@ -82,3 +82,10 @@ impl<T: PeerAddr, S> PeerAddr for SensorIo<T, S> {
         self.io.peer_addr()
     }
 }
+
+#[async_trait::async_trait]
+impl<I: Peek + Send + Sync, S: Sensor + Sync> Peek for SensorIo<I, S> {
+    async fn peek(&self, buf: &mut [u8]) -> Result<usize> {
+        self.io.peek(buf).await
+    }
+}
```

Comment:
Had to add that, because The TLS stack requires the IO type to be peekable. If there is an easy way to avoid that, it would be great.